### PR TITLE
[Don't merge] Fix reverse calc for trade with max dest amount

### PR DESCRIPTION
**Pull request by manhlx3006**
Experiment fixing reverse calc for trade with max dest amount
Please pull and run tests in kyberNetwork.js to see some tests with maxDestAmount + split are failing

### CODE REVIEW

File: contractsSol5/KyberNetwork.sol [hunk 1]
```diff
@@ -729,15 +729,13 @@ contract KyberNetwork is Withdrawable2, Utils4, IKyberNetwork, ReentrancyGuard {
...
     {
-        uint destAmountSoFar;
-
-        for (uint i = 0; i < rates.length; i++) {
-            uint destAmountSplit = i == (splitValuesBps.length - 1) ?
-                (destAmount - destAmountSoFar) : splitValuesBps[i] * destAmount / BPS;
-            destAmountSoFar += destAmountSplit;
-
-            srcAmount += calcSrcQty(destAmountSplit, srcDecimals, destDecimals, rates[i]);
+        uint averageRate = 0;
```
Comment (ilanDoron) [inline contractsSol5/KyberNetwork.sol:732]: should avoid this init.

File: contractsSol5/KyberNetwork.sol [hunk 2]
```diff
@@ -729,15 +729,13 @@ contract KyberNetwork is Withdrawable2, Utils4, IKyberNetwork, ReentrancyGuard {
...
+        for(uint i = 0; i < rates.length; i++) {
+            averageRate += splitValuesBps[i] * rates[i] / BPS;
         }
+        require(averageRate > 0, "average rate is 0");
```
Comment (ilanDoron) [inline contractsSol5/KyberNetwork.sol:736]: we can also calculate average rate from 
ActualDestAmount and actual srcAmount

Comment (Anyhowclick) [reply contractsSol5/KyberNetwork.sol:736]: Yeah, using just `calcRateFromQty` might be enough

**Comment by manhlx3006** [comment]
Will make a fix to katalystRevamp branch

**Pull request closed without merging**